extensions/identity-hub/src/main/java/org/eclipse/dataspaceconnector/identityhub/processor/CollectionsWriteProcessor.java

```diff
@@ -0,0 +1,38 @@
+package org.eclipse.dataspaceconnector.identityhub.processor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.dataspaceconnector.identityhub.dtos.HubObject;
+import org.eclipse.dataspaceconnector.identityhub.dtos.MessageResponseObject;
+import org.eclipse.dataspaceconnector.identityhub.dtos.MessageStatus;
+import org.eclipse.dataspaceconnector.identityhub.dtos.VerifiableCredential;
+import org.eclipse.dataspaceconnector.identityhub.store.IdentityHubStore;
+
+import java.util.Base64;
+
+import static org.eclipse.dataspaceconnector.identityhub.dtos.MessageResponseObject.MESSAGE_ID_VALUE;
+
+/**
+ * Processor of "CollectionsWrite" messages, in order to write {@link HubObject}s into the {@link IdentityHubStore}.
+ */
+public class CollectionsWriteProcessor implements MessageProcessor {
+
+    private IdentityHubStore identityHubStore;
+    private ObjectMapper mapper;
+
+    public CollectionsWriteProcessor(IdentityHubStore identityHubStore) {
+        this.identityHubStore = identityHubStore;
+        this.mapper = new ObjectMapper();
+    }
+
+    public MessageResponseObject process(byte[] data) {
+        try {
+            var decodedData = new String(Base64.getUrlDecoder().decode(data));
```

Comment:
I think it does not work without registering some custom deserializers in Jackson. The "data" field is a base64url encoded string as [per spec](https://identity.foundation/decentralized-web-node/spec/#messages):

`Message objects MAY contain a data property, and if present its value MUST be a base64Url encoded string of the Message’s data.`

Reply:
I see, I'd missed the url encoding part. Probably we could still use Jackson if we want to, instead of custom code to deser. Not sure what's the best way

Reply:
Honestly I would leave it like it is for now. Its one line of code VS probably a class for Jackson deserialization. We can improve as we go if we see a better way.

Reply:
Sure, but are we sure we are not doing double-decoding with the current process?